src/test/java/Drivers/WebDriverFactory.java

```diff
@@ -0,0 +1,15 @@
+package Drivers;
+
+import Enums.BrowserName;
+import org.openqa.selenium.WebDriver;
+
+public class WebDriverFactory {
+    public static WebDriver getWebDriver(BrowserName browserName) {
+        return switch (browserName) {
+            case Chrome -> new ChromeDriverManager().getDriver();
+            case Firefox -> new FirefoxDriverManager().getDriver();
+            default ->
+                    throw new UnsupportedOperationException("This method is not implemented yet"); //to discuss what exact Exception this should throw
```

Comment:
It might be beneficial to define a custom exception that indicates that a browser type is unsupported, rather than using the generic `UnsupportedOperationException`.
```java
public class UnsupportedBrowserException extends Exception {
    public UnsupportedBrowserException() {
         super("Automation  for browser" + browserName + " is not implemented yet");
   }
}
```

Reply:
@fnfldx To be discussed whether a class specifically for this kind of exception has to be implemented.

Reply:
@fnfldx Added new class in the latest commit for throwing a custom exception.